tests/ui/doc.rs

```diff
@@ -203,6 +203,12 @@ fn issue_2343() {}
 /// __|_ _|__||_|
 fn pulldown_cmark_crash() {}
 
+/// This is a doc comment with `unbalanced_tick marks and several words that
+/// should be `encompassed_by` tick marks because they `contain_underscores`.
+/// Because of the initial `unbalanced_tick` pair, the error message is
+/// very `confusing_and_misleading`.
+fn issue_6753() {}
```

Comment:
Could you also add a test that has multiple Markdown paragraphs and headlines etc.

And one that has a backtick inside a code block like this:
```
// Moo - I'm a cow
` <-- notice the single backtick
// some more text
```